### PR TITLE
Fix maths

### PR DESCRIPTION
**Pull request by swapdisk**
Fixed issues raised under #58 and generally cleaned up and simplified the `bigboot.sh` script. No changes required to existing playbooks. Working as expected in my initial testing, but I will keep banging on it, so opening the PR as draft for now.

### CODE REVIEW

File: roles/bigboot/files/bigboot.sh [hunk 1]
```diff
@@ -249,12 +249,10 @@ convert_size_to_fs_blocks() {
 }
 
 deactivate_volume_group() {
-    local volume_group_name
-    volume_group_name=$(get_volume_group_name)
-    ret=$(/usr/sbin/lvm vgchange -an "$volume_group_name" 2>&1)
+    ret=$(/usr/sbin/lvm vgchange -an "$LVM2_VG_NAME" 2>&1)
```
Comment (ygalblum) [inline roles/bigboot/files/bigboot.sh:252]: You're not using `get_volume_group_name`, is it used somewhere else? If no, please remove it. If yes, why use two methods for the same information?

File: roles/bigboot/files/bigboot.sh [hunk 2]
```diff
@@ -263,30 +261,24 @@ deactivate_volume_group() {
 
 check_available_free_space() {
     local device="${DEVICE_NAME}${PARTITION_PREFIX}${ADJACENT_PARTITION_NUMBER}"
-    free_device_space_in_bytes=$(get_free_device_size)
-    # if there is enough free space after the adjacent partition, there is no need to shrink it.
-    if [[ $free_device_space_in_bytes -gt $INCREMENT_BOOT_PARTITION_SIZE_IN_BYTES ]]; then
-        SHRINK_SIZE_IN_BYTES=0
-        return
+    # Get LVM details
+    eval "$(/usr/sbin/lvm pvs --noheadings --nameprefixes --nosuffix --units b -o vg_name,vg_extent_size,pv_pe_count,pv_pe_alloc_count,vg_free_count $device)"
```
Comment (ygalblum) [inline roles/bigboot/files/bigboot.sh]: I'm probably missing something here. What are you getting out of this call?

**Review by ygalblum — COMMENTED** [review]
Also please squash these commit (at least the ones that are back and forth of the same change)

File: roles/bigboot/files/bigboot.sh (hunk 1 above)
Comment (swapdisk) [reply roles/bigboot/files/bigboot.sh:245]: Right, `get_volume_group_name` in not used anywhere and I removed it. All of the LVM info is now populated with this line in `check_available_free_space`...
```
eval "$(/usr/sbin/lvm pvs --noheadings --nameprefixes --nosuffix --units b -o vg_name,vg_extent_size,pv_pe_count,pv_pe_alloc_count,vg_free_count "$device")"
```

File: roles/bigboot/files/bigboot.sh (hunk 2 above)
Comment (swapdisk) [reply roles/bigboot/files/bigboot.sh]: It sets all the LVM info variables we need with one command.

Comment (ygalblum) [reply roles/bigboot/files/bigboot.sh]: IIUC this command displays information about the PV, but the information is not saved or used anywhere afterwards. Is it?

Comment (swapdisk) [reply roles/bigboot/files/bigboot.sh]: > IIUC this command displays information about the PV, but the information is not saved or used anywhere afterwards. Is it?

It is. The output of the `pvs` command would be something like this example:

```
# pvs --noheadings --nameprefixes --nosuffix --units b -o vg_name,vg_extent_size,pv_pe_count,pv_pe_alloc_count,vg_free_count /dev/nvme0n1p2 
  LVM2_VG_NAME='VolGroup00' LVM2_VG_EXTENT_SIZE='4194304' LVM2_PV_PE_COUNT='8072' LVM2_PV_PE_ALLOC_COUNT='5120' LVM2_VG_FREE_COUNT='2952'
```

By using the command substitution and `eval` builtin, bash sets variables using the names and values output by the `pvs` command. Here's an example to demonstrate some of the vars being set:

```
# device=/dev/nvme0n1p2
# eval "$(/usr/sbin/lvm pvs --noheadings --nameprefixes --nosuffix --units b -o vg_name,vg_extent_size,pv_pe_count,pv_pe_alloc_count,vg_free_count $device)"
# echo $LVM2_VG_NAME
VolGroup00
# echo $LVM2_VG_EXTENT_SIZE
4194304
```

Comment (ygalblum) [reply roles/bigboot/files/bigboot.sh]: Interesting. I didn't know that. Thanks for the explanation

**Comment by swapdisk** [comment]
> Also please squash these commit (at least the ones that are back and forth of the same change)

Done.

**Review by ygalblum — APPROVED** [review]
LGTM

**Pull request merged**